src/runtime/auto-prompt-manager.ts

```diff
@@ -494,60 +512,26 @@ export class AutoPromptManager {
       )
     );
 
-    // No audience actions means use the default prompt, if it should be shown.
-    if (potentialActions.length === 0) {
-      if (shouldShowMonetizationPromptAsSoftPaywall) {
-        this.interventionDisplayed_ = this.isSubscription_({autoPromptType})
-          ? {type: TYPE_SUBSCRIPTION}
-          : this.isContribution_({autoPromptType})
-          ? {type: TYPE_CONTRIBUTION}
-          : null;
-      }
-      return undefined;
-    }
-
-    // For subscriptions, skip triggering checks and use the first potential action
-    if (this.isSubscription_({autoPromptType})) {
-      if (
-        shouldShowMonetizationPromptAsSoftPaywall ||
-        potentialActions[0].type === TYPE_SUBSCRIPTION
-      ) {
-        this.interventionDisplayed_ = {type: TYPE_SUBSCRIPTION};
-        return undefined;
-      }
-      const firstAction = potentialActions[0];
-      this.interventionDisplayed_ = firstAction;
-      return firstAction;
-    }
-
-    // Suppress previously dismissed prompts.
-    let previouslyShownPrompts: string[] = [];
-    if (dismissedPrompts) {
-      previouslyShownPrompts = dismissedPrompts.split(',');
+    if (!this.isSubscription_(autoPromptType) && !this.pageConfig_.isLocked()) {
+      // Filter out contribution prompt if page is not paywalled.
       potentialActions = potentialActions.filter(
-        (action) => !previouslyShownPrompts.includes(action.type)
+        (action) => action.type !== TYPE_CONTRIBUTION
       );
-    }
 
-    const contributionIndex = potentialActions.findIndex(
-      (action) => action.type === TYPE_CONTRIBUTION
-    );
-    // If autoprompt should be shown, and the contribution action is either the first action or
-    // not passed through audience actions, honor it and display the contribution prompt.
-    if (shouldShowMonetizationPromptAsSoftPaywall && contributionIndex < 1) {
-      this.interventionDisplayed_ = {type: TYPE_CONTRIBUTION};
-      return undefined;
+      // Suppress previously dismissed prompts.
+      let previouslyShownPrompts: string[] = [];
+      if (dismissedPrompts) {
+        previouslyShownPrompts = dismissedPrompts.split(',');
```

Comment:
nit: `const previouslyShownPrompts` as it's used within the if statement scope.

Reply:
Good catch.